### PR TITLE
added test coverage for LDAP auth server with HTTPS protocol

### PR DESCRIPTION
**Pull request by omkarkhatavkar**
### PR Objective

- Refactored the LDAP tests
- Resolved the test failures from LDAP tests
- Added the ldap_wrapper decorator to manage the teardown for each test.
- Made Changes in ordering and removing the unwanted call to the user.update 
- Removed the Bugzilla as this test is passing if the hostname is passed instead of the IP address.
     

### Test Result 

```
(env) [root@okhatavk robottelo]# pytest tests/foreman/ui/test_ldap_authentication.py 
============================================= test session starts ==============================================
platform linux -- Python 3.7.2, pytest-4.6.3, py-1.8.0, pluggy-0.13.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/okhatavk/Satellite/robottelo
plugins: cov-2.8.1, xdist-1.30.0, mock-1.10.4, services-1.3.1, forked-1.1.1
collecting ... 2019-12-24 09:37:40 - conftest - DEBUG - Collected 16 test cases

2019-12-24 15:07:50 - robottelo.ssh - DEBUG - Destroyed Paramiko client 0x7f0d1154a898
2019-12-24 15:07:50 - robottelo.host_info - DEBUG - Host Satellite version: 6.7
2019-12-24 15:07:50 - robottelo.helpers - DEBUG - Generated file bz_cache.json with BZ collect data
collected 16 items                                                                                             

tests/foreman/ui/test_ldap_authentication.py ................                                            [100%]

=============================================== warnings summary ===============================================
tests/foreman/ui/test_ldap_authentication.py::test_positive_end_to_end_ad
tests/foreman/ui/test_ldap_authentication.py::test_positive_end_to_end_ad
  /home/okhatavk/Satellite/robottelo/env/lib/python3.7/site-packages/widgetastic/browser.py:721: DeprecationWarning: use driver.switch_to.alert instead
    return self.selenium.switch_to_alert()

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================================== 16 passed, 2 warnings in 3712.19 seconds ===================================

```

### CODE REVIEW

File: tests/foreman/ui/test_ldap_authentication.py [hunk 1]
```diff
@@ -330,6 +334,82 @@ def test_positive_create_with_idm_org_and_loc(session, ipa_data):
...
+            session.organization.select(org_name=org.name)
+            session.location.select(loc_name=loc.name)
+            assert session.ldapauthentication.read_table_row(
+                ldap_auth_name)['Name'] == ldap_auth_name
+            ldap_source = session.ldapauthentication.read(ldap_auth_name)
+            assert ldap_source['ldap_server']['name'] == ldap_auth_name
+            assert ldap_source[
+                'ldap_server']['host'] == ipa_data['ldap_ipa_hostname']
+            assert ldap_source['ldap_server']['port'] == '636'
+        username = 'foreman'
```
Comment (ntkathole) [inline tests/foreman/ui/test_ldap_authentication.py]: settings.ipa.user_ipa ?

File: tests/foreman/ui/test_ldap_authentication.py [hunk 2]
```diff
@@ -330,6 +334,82 @@ def test_positive_create_with_idm_org_and_loc(session, ipa_data):
...
+            session.location.select(loc_name=loc.name)
+            assert session.ldapauthentication.read_table_row(
+                ldap_auth_name)['Name'] == ldap_auth_name
+            ldap_source = session.ldapauthentication.read(ldap_auth_name)
+            assert ldap_source['ldap_server']['name'] == ldap_auth_name
+            assert ldap_source[
+                'ldap_server']['host'] == ipa_data['ldap_ipa_hostname']
+            assert ldap_source['ldap_server']['port'] == '636'
+        username = settings.ipa.user_ipa
+        full_name = f'{settings.ipa.user_ipa} katello'
```
Comment (ntkathole) [inline tests/foreman/ui/test_ldap_authentication.py]: remove f-string as Betelgeuse is not compatible with it.

Comment (omkarkhatavkar) [reply tests/foreman/ui/test_ldap_authentication.py]: ohh okay

**Comment by omkarkhatavkar** [comment]
I have made changes gitlab PR, We should be good to merge this PR once if that PR looks okay.

**Comment by vijay8451** [comment]
@omkarkhatavkar MR was `293` , right ?

**Comment by omkarkhatavkar** [comment]
> @omkarkhatavkar MR was `293` , right ?

Yes It is merged

**Pull request merged**